rhel.Dockerfile

```diff
@@ -0,0 +1,129 @@
+#
+# Copyright (c) 2018-2019 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+#
+
+# Builder: check meta.yamls and create index.json
+# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8-minimal
+FROM registry.access.redhat.com/ubi8-minimal:8.0-213 as builder
+USER 0
+
+################# 
+# PHASE ONE: create ubi8-minimal image with yq
+################# 
+
+# to get all the python deps pre-fetched so we can build in Brew:
+# 1. extract files in the container to your local filesystem
+#    CONTAINERNAME="pluginregistry"
+#    mkdir -p /tmp/root-local/ && docker run -it -v /tmp/root-local/:/tmp/root-local/ ${CONTAINERNAME} /bin/bash -c "cd /root/.local/ && cp -r bin/ lib/ /tmp/root-local/"
+# 2. then back on your own machine, tar them up and put them in this folder
+#    cd /tmp/root-local && sudo tar cvzf root-local.tgz lib/ bin/ && cd - && mv -f /tmp/root-local/root-local.tgz . && sudo rm -fr /tmp/root-local/
+# 3. then add it to dist-git so it's part of this repo
+#    rhpkg new-sources root-local.tgz 
+
+# built in Brew, use tarball in lookaside cache
+# COPY root-local.tgz /tmp/root-local.tgz
+
+# NOTE: uncomment for local build. Must also set full registry path in FROM to registry.redhat.io or registry.access.redhat.com
+# enable rhel 7 or 8 content sets (from Brew) to resolve jq as rpm
+COPY content_sets_epel7.repo /etc/yum.repos.d/
+
+RUN microdnf install -y findutils bash wget yum gzip tar jq python3-six python3-pip && microdnf -y clean all && \
+    # install yq (depends on jq and pyyaml - if jq and pyyaml not already installed, this will try to compile it)
+    if [[ -f /tmp/root-local.tgz ]]; then mkdir -p /root/.local && tar xf /tmp/root-local.tgz -C /root/.local/ && rm -fr /tmp/root-local.tgz; fi && \
+    /usr/bin/pip3.6 install --user yq jsonschema && \
```

Comment:
I believe that if you don't use the `--user` install, you don't need the next "find-and-copy-into-$PATH" step.

Reply:
IIRC pip doesn't like to run as root so the --user flag was required here. Prove me wrong?

Reply:
`pip` doesn't like to run as root in general since you can break stuff, but I don't know that the user flag helps a whole lot when your user is root (I still get the 
> WARNING: Running pip install with root privileges is generally not a good idea. Try `pip3.6 install --user` instead.

warning with the flag).

The dockerfile
```dockerfile
FROM registry.access.redhat.com/ubi8-minimal:8.0-213
USER 0

COPY content_sets_epel7.repo /etc/yum.repos.d/
RUN microdnf install -y findutils jq python3-six python3-pip && microdnf -y clean all

RUN /usr/bin/pip3.6 install yq

RUN echo '{"test": "hello"}' | yq '.'
```
works for me.

Reply:
Yes but YOU have access to the public internet. Brew does not. So I install from the `root-local.tgz` instead of fetching from `https://files.pythonhosted.org/packages` for the whl and tar.gz files for dependencies. That's why my approach is more convoluted (but can fall back to fetching from the internet if the tarball isn't present).

Perhaps I should change the logic such that it omits the `--user` flag and all the `cp` commands if the tarball isn't found?

Reply:
OK, I've updated the logic so that it only installs to the /root/.local folder if BOOTSTRAP=true or there's a tarball copied into the container.